cmd/storage-rpc-client.go

```diff
@@ -23,18 +23,14 @@ import (
 	"net/rpc"
 	"net/url"
 	"path"
-	"sync"
 	"sync/atomic"
-	"time"
 
 	"github.com/minio/minio/pkg/disk"
 )
 
 type networkStorage struct {
+	rpcClient         *AuthRPCClient
```

Comment:
You can't inherit from AuthRPC - we need a separate implementation to make sure that the storage client doesn't connect and start initiating calls directly to server. 

Will send a PR to your PR incorporating those changes.